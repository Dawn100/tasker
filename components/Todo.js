import React, { Component } from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      style: this.props.todo.completed ? styles.done : styles.waiting
    };
  }

  complete = () => {
    this.setState({ completed: true });
  };
  render() {
    return (
      <TouchableOpacity style={this.state.style} onPress={this.complete}>
        <View testID={"root"} style={{ backgroundColor: "white" }}>
          <Text testID={"name"}>{this.props.todo.todo}</Text>
          <Text testID={"status"}>
            Is complete? {this.props.todo.completed + ""}
          </Text>
          <Text testID={"id"}>{this.props.todo.id}</Text>
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  waiting: {
    color: "#0f0f0f",
    backgroundColor: "#ccc"
  },
  done: {
    color: "#f0f0f0",
    backgroundColor: "#008080"
  }
});
export default Todo;
