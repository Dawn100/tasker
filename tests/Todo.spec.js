import { create } from "react-test-renderer";
import React from "react";
import Todo from "../components/Todo";
import { mount } from "enzyme";

describe("Todo", () => {
  const testTodo = {
    id: 0,
    todo: "Todo 0",
    completed: false
  };
  const todo = create(<Todo todo={testTodo} />);
  it("should render self and subcomponents", () => {
    const todoJSON = todo.toJSON();
    expect(todoJSON).toMatchSnapshot();
    console.log(todoJSON);

    expect(todoJSON.type).toEqual("View");

    expect(todoJSON.children[0].children[0].type).toEqual("Text");
    expect(todoJSON.children[0].children[0].children[0]).toEqual("Todo 0");

    expect(todoJSON.children[0].children[1].type).toEqual("Text");
    expect(todoJSON.children[0].children[1].children[0]).toEqual(
      "Is complete? ",
      "false"
    );

    expect(todoJSON.children[0].children[2].type).toEqual("Text");
    expect(todoJSON.children[0].children[2].children[0]).toEqual("0");
  });
  it("it  has right state", () => {
    const todoInstance = todo.getInstance();
    // console.log(todoInstance);
  });
});
