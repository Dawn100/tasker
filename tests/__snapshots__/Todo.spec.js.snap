// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should render self and subcomponents 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ccc",
      "color": "#0f0f0f",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
    testID="root"
  >
    <Text
      testID="name"
    >
      Todo 0
    </Text>
    <Text
      testID="status"
    >
      Is complete? 
      false
    </Text>
    <Text
      testID="id"
    >
      0
    </Text>
  </View>
</View>
`;
